lightning/src/ln/channelmanager.rs

```diff
@@ -6774,6 +6836,32 @@ impl<'a, Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 			}
 		}
 
+		let mut outbound_scid_aliases = HashSet::new();
+		for (chan_id, chan) in by_id.iter_mut() {
+			if chan.outbound_scid_alias() == 0 {
+				let mut outbound_scid_alias;
+				loop {
+					outbound_scid_alias = fake_scid::Namespace::OutboundAlias
+						.get_fake_scid(best_block_height, &genesis_hash, fake_scid_rand_bytes.as_ref().unwrap(), &args.keys_manager);
+					if outbound_scid_aliases.insert(outbound_scid_alias) { break; }
+				}
+				chan.set_outbound_scid_alias(outbound_scid_alias);
+			} else if !outbound_scid_aliases.insert(chan.outbound_scid_alias()) {
+				// Note that in rare cases its possible to hit this while reading an older
+				// channel if we just happened to pick a colliding outbound alias above.
+				log_error!(args.logger, "Got duplicate outbound SCID alias; {}", chan.outbound_scid_alias());
+				return Err(DecodeError::InvalidValue);
+			}
+			if chan.is_usable() {
+				if short_to_id.insert(chan.outbound_scid_alias(), *chan_id).is_some() {
```

Comment:
Do we intend to have two entries for channels with an `outbound_alias` in `short_to_id`? Could we add some docs for `short_to_id` and mention aliases?